specs/deployment.md

```diff
@@ -1,599 +1,166 @@
-# Paste S3 bucket name (e.g., my-second-brain-backup)
-
-wrangler secret put S3_BACKUP_REGION
-# Paste AWS region (e.g., us-east-1)
-```
-
-### 6. Set Development Secrets (Optional)
+**Storage isolation** requires separate R2 buckets for user file storage, ensuring development testing cannot corrupt or expose production user data. Development buckets SHOULD use naming conventions that prevent accidental production operations (e.g., `second-brain-dev` vs `second-brain`).
 
-```bash
-# Same as production but for dev environment
-wrangler secret put GITHUB_CLIENT_ID --env development
-wrangler secret put GITHUB_CLIENT_SECRET --env development
-wrangler secret put COOKIE_ENCRYPTION_KEY --env development
-# ... etc
-```
+**Session isolation** requires separate Durable Object namespaces, preventing development session IDs from colliding with production sessions. Each environment's Durable Objects MUST use distinct class names or namespace bindings.
 
----
-
-## Deployment
+**Authentication isolation** requires separate KV namespaces for OAuth tokens and rate limiting state. Development and production MUST use different OAuth client credentials to prevent token reuse across environments.
 
-### Run Tests
+**Analytics isolation** SHOULD separate development and production telemetry to prevent test traffic from skewing production metrics. Cloudflare Analytics Engine SHOULD be configured with environment-specific dataset names.
 
-```bash
-pnpm test
-```
+### Environment Parity
 
-Ensure all tests pass before deploying.
+While isolated, both environments MUST maintain configuration parity to ensure changes validated in development behave identically in production. Parity applies to:
 
-### Deploy to Development
+**Infrastructure:** Identical Worker configuration including compatibility dates, CPU limits, and binding types. Both environments MUST use the same wrangler.toml structure with only environment-specific values differing.
 
-```bash
-wrangler deploy --env development
-```
+**Secrets:** Both environments require the same secret keys (GitHub OAuth credentials, cookie encryption keys, S3 backup credentials) though the actual secret values MAY differ. Development MAY use test credentials where appropriate.
 
-Note the deployed URL (e.g., `https://second-brain-mcp.dev.your-subdomain.workers.dev`)
+**Dependencies:** Package versions, runtime configuration, and feature flags MUST match. The development environment MUST NOT enable experimental features or configurations not intended for production.
 
-### Test Development Deployment
+### Environment URLs
 
-1. Configure Claude to connect to dev URL
-2. Complete OAuth flow
-3. Test all tools (read, write, edit, glob, grep)
-4. Verify bootstrap files created
-5. Check rate limiting
-6. Test backup trigger (if configured)
+Each environment MUST be accessible via a stable HTTPS URL:
 
-### Deploy to Production
+- Development: `https://second-brain.{username}.workers.dev` (Workers subdomain)
```

Comment:
This is not right. I think it might actually be dev.second-brain. If you're unable to verify then just leave this needing clarification.